File: Leetcode/1.两数之和.kt
/*
 * @lc app=leetcode.cn id=1 lang=kotlin
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var indexFirst = 0
        var indexSecond = 1
        for (a in nums.indices) {
            for (b in nums.indices) {
                if (a != b && nums[a] + nums[b] == target) {
                    indexFirst = a
                    indexSecond = b
                    break
                }
            }
        }
        if (indexFirst > indexSecond) {
            val temp = indexFirst
            indexFirst = indexSecond
            indexSecond = temp
        }
        return intArrayOf(indexFirst, indexSecond)
        //当一个值只在返回的时候使用的话使用内联函数的写法
    }
}
// @lc code=end

File: Leetcode/1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

//date: 2021-10-03

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let hash = new Map();

    for (let i = 0; i < nums.length; i++) {
        let another = target - nums[i];
        if (hash.has(another)) {
            return [hash.get(another), i];
        }
        hash.set(nums[i], i);
    }
};
// @lc code=end
